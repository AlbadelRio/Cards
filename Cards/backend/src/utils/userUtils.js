function dataChanges(body, userData, userFound) {
  switch (body.type) {
    case 'cardsAdquired':
      userData.cardsAdquired.push(body.cardId);
      userFound.save();
      break;

    case 'difficultCards':
      userData.difficultCards.push(body.cardId);
      userFound.save();
      break;
    default:
      break;
  }
}

module.exports = { dataChanges };
