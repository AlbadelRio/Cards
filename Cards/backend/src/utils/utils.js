function DataChanges(body, userData, userFound) {
  switch (body.type) {
    case 'cardsAdquired':
      userData.cardsAdquired.push(body.cardId);
      userFound.save();
      break;

    case 'difficultCards':
      userData.difficultCards.push(body.cardId);
      userFound.save();
      break;
    case 'stats': {
      const existingDate = userData.stats.find((data) => data.date === body.date);
      if (!existingDate) {
        userData.stats.push(body);
        userFound.save();
      } else {
        existingDate.stats.update(
          {
            $push: { timePlayed: body.time, percentages: body.percentage },
            $sort: { score: -1 }
          }

        );
      } }
      break;
    default:
      break;
  }
}

module.exports = { DataChanges };
