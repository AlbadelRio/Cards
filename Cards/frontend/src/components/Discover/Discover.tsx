/* eslint-disable no-underscore-dangle */
import React, { useEffect } from 'react';
import {
  View, Text
} from 'react-native';
import { useDispatch, useSelector } from 'react-redux';
import { loadPackcards } from '../../redux/actions/packCardsActionCreators';
import styles from './discoverStlyles';

export default function Discover() {
  const { token, refreshToken } = useSelector((store:any) => store.tokensReducer);
  const packCards = useSelector((store:any) => store.packardsReducer);

  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(loadPackcards(token, refreshToken));
  }, []);

  const filteredPackCard = Array.from(new Set(packCards.map((packCard:any) => packCard.subject)))
    .map((subject) => ({
      subject
    }));

  return (

    <View
      style={{ flex: 2 }}
    >
      <View
        style={{ flex: 1 }}
      >
        <Text
          style={styles.title}
        >
          DISCOVER

        </Text>
        <Text
          style={styles.subtitle}
        >
          Choose from these categories what interets you the most

        </Text>
      </View>
      <View
        style={styles.containerList}
      >
        {filteredPackCard.map((element:any) => (
          <Text
            key={element._id}
            style={styles.subjects}

          >
            {element.subject}
          </Text>
        ))}

      </View>
    </View>
  );
}
