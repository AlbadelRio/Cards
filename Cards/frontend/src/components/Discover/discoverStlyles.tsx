import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  title: {
    fontSize: 50,
    color: '#222222',
    textAlign: 'center',
    fontWeight: 'bold',
    marginTop: 40
  },
  subtitle: {
    color: '#222222',
    textAlign: 'center',
    marginTop: 20
  },
  containerList: {

    marginTop:5,
    width: 500,
    height: 300,
    backgroundColor: 'red',
    flexDirection: 'row',
    alignContent: 'space-between',
    flexWrap: 'wrap'

  },
  subjects: {
    margin: 8,
    padding: 10,
    width: 110,
    height: 40,
    backgroundColor: '#222222',
    color: '#c5c5c5',
    fontWeight: 'bold',
    textAlign: 'center',
    flexWrap: 'wrap'
  }
});
