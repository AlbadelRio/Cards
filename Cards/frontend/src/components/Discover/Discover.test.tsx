import React from 'react';
import { render, fireEvent } from '../../utils/test.utils';

import Discover from './Discover';

const navigation = { navigte: jest.fn() };

describe('Given a Discover component', () => {
  describe('when it is rendered', () => {
    let screen:any;
    beforeEach(() => {
      const initialState = {
        packCards: [{ subject: 'abc' }],
        randomPack: [{ subject: 'abc' }],
        token: 'hddjjdj',
        refreshToken: 'dkkdkd'
      };

      screen = render(<Discover navigation={navigation} />, initialState);
    });
    test('Then should match the snapshot', () => {
      expect(screen).toMatchSnapshot();
    });
    describe('When pressable is pressed', () => {
      test.only('the randomHandler has to been called', () => {
        const filteredPackCard = [{ subject: 'abc' }];

        const randomHandler = jest.fn();
        const pressableItem = screen.getByText('abc');
        fireEvent.press(pressableItem);
        expect(screen.queryByTestId(randomHandler)).toHaveBeenCalled();
      });
    });
  });
});
