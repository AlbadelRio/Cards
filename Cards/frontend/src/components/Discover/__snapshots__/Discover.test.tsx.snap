// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a Discover component when it is rendered Then should match the snapshot 1`] = `
<BVLinearGradient
  colors={
    Array [
      4284398512,
      4284309487,
    ]
  }
  endPoint={
    Object {
      "x": 0,
      "y": 1,
    }
  }
  locations={null}
  startPoint={
    Object {
      "x": 0,
      "y": 0,
    }
  }
  style={
    Object {
      "height": "100%",
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "flex": 2,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#222222",
            "fontSize": 50,
            "fontWeight": "bold",
            "marginTop": 40,
            "textAlign": "center",
          }
        }
      >
        DISCOVER
      </Text>
      <Text
        style={
          Object {
            "color": "#222222",
            "textAlign": "center",
          }
        }
      >
        Choose from these categories what interets you the most
      </Text>
    </View>
    <RCTScrollView
      horizontal={true}
    >
      <View>
        <View
          style={
            Object {
              "flexDirection": "row",
              "flexWrap": "wrap",
              "height": 100,
              "justifyContent": "space-evenly",
              "width": 510,
            }
          }
        />
      </View>
    </RCTScrollView>
  </View>
</BVLinearGradient>
`;
